src/frontend/screens/Settings/MapManagement/CustomMapDetails.tsx

```diff
@@ -0,0 +1,153 @@
+import {TouchableOpacity} from 'react-native-gesture-handler';
+
+import {bytesToMegabytes} from '../../../lib/bytesToMegabytes';
+import {BLACK, MEDIUM_GREY, RED, VERY_LIGHT_GREY} from '../../../lib/styles';
+import {Loading} from '../../../sharedComponents/Loading';
+import {Text} from '../../../sharedComponents/Text';
+
+const m = defineMessages({
+  mapNameColumn: {
+    id: 'screens.Settings.MapManagement.MapsList.CustomMapDetails.mapNameColumn',
+    defaultMessage: 'Map Name',
+  },
+  dateAddedColumn: {
+    id: 'screens.Settings.MapManagement.MapsList.CustomMapDetails.dateAdded',
+    defaultMessage: 'Date Added',
+  },
+  removeMap: {
+    id: 'screens.Settings.MapManagement.MapsList.CustomMapDetails.removeMap',
+    defaultMessage: 'Remove Map',
+  },
+  sizeInMegabytes: {
+    id: 'screens.Settings.MapManagement.MapsList.CustomMapDetails.sizeInMegabytes',
+    defaultMessage: '{value} MB',
+  },
+});
+
+export function CustomMapDetails({
+  dateAdded,
+  loading,
+  name,
+  onRemove,
+  size,
+}: {
+  dateAdded: Date;
+  loading?: boolean;
+  name: string;
+  onRemove: () => void;
+  size?: number;
+}) {
+  const {formatMessage: t} = useIntl();
+
+  const calculatedSize = size ? bytesToMegabytes(size).toFixed(0) : undefined;
+  const displayedSize =
+    calculatedSize === undefined
+      ? undefined
+      : parseInt(calculatedSize, 10) < 1
+        ? '<1'
+        : calculatedSize;
+
+  return (
+    <View style={styles.rootContainer}>
+      <View style={styles.columnHeadersContainer}>
+        <Text style={styles.columnTitleText}>{t(m.mapNameColumn)}</Text>
+        <Text style={styles.columnTitleText}>{t(m.dateAddedColumn)}</Text>
+      </View>
+      <View style={styles.cardContainer}>
+        <View style={styles.cardRow}>
+          <View style={styles.columnLeft}>
+            <Text style={styles.nameText}>{name}</Text>
+            <Text style={styles.sizeText}>
+              {displayedSize !== undefined &&
+                t(m.sizeInMegabytes, {
+                  value: displayedSize,
+                })}
+            </Text>
+          </View>
+          <View style={styles.columnRight}>
+            <Text style={styles.dateAddedText}>
+              <FormattedDate
+                year="numeric"
+                month="short"
+                day="2-digit"
+                value={dateAdded}
+              />
+            </Text>
+          </View>
+        </View>
+        <View style={styles.cardRow}>
+          <TouchableOpacity onPress={onRemove} hitSlop={20}>
+            <Text style={styles.removeMapText}>{t(m.removeMap)}</Text>
+          </TouchableOpacity>
+        </View>
+        {loading && <LoadingOverlay />}
+      </View>
+    </View>
+  );
+}
+
+function LoadingOverlay() {
+  return (
+    <View pointerEvents="none" style={styles.overlay}>
+      <View style={[styles.overlay, {opacity: 0.1, backgroundColor: BLACK}]} />
+      <Loading size={10} />
+    </View>
+  );
+}
```

Comment:
could be useful to extract as a shared component, but would prefer to avoid the extra work if not necessary for now